homeassistant/components/sensorpush_cloud/sensor.py

```diff
@@ -0,0 +1,162 @@
+    ATTR_SIGNAL_STRENGTH,
+    ATTR_VAPOR_PRESSURE,
+    DOMAIN,
+    LOGGER,
+    MAX_TIME_BETWEEN_UPDATES,
+)
+from .coordinator import SensorPushCloudConfigEntry, SensorPushCloudCoordinator
+
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+SENSORS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key=ATTR_ALTITUDE,
+        device_class=SensorDeviceClass.DISTANCE,
+        entity_registry_enabled_default=False,
+        translation_key="altitude",
+        native_unit_of_measurement=UnitOfLength.FEET,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_ATMOSPHERIC_PRESSURE,
+        device_class=SensorDeviceClass.ATMOSPHERIC_PRESSURE,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfPressure.INHG,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_BATTERY_VOLTAGE,
+        device_class=SensorDeviceClass.VOLTAGE,
+        entity_registry_enabled_default=False,
+        translation_key="battery_voltage",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_DEWPOINT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_registry_enabled_default=False,
+        translation_key="dewpoint",
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_HUMIDITY,
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_SIGNAL_STRENGTH,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_TEMPERATURE,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_VAPOR_PRESSURE,
+        device_class=SensorDeviceClass.PRESSURE,
+        entity_registry_enabled_default=False,
+        translation_key="vapor_pressure",
+        native_unit_of_measurement=UnitOfPressure.KPA,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SensorPushCloudConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SensorPush Cloud sensors."""
+    coordinator: SensorPushCloudCoordinator = entry.runtime_data
+    device_ids = await coordinator.async_get_device_ids()
```

Comment:
I'd rather do that in `_async_setup` of the coordinator (which gets called during `async_config_entry_first_refresh`) and store it in the coordinator, so we don't have to do IO again

Reply:
The thought at the time was to handle configuring additional entities without requiring a restart. The list returned by the API isn't fixed and is populated based on actions taken on the mobile app.

Reply:
That'll require a bit more work, since this is only called once. You'll need to call this during a regular update and register callback listeners to add the new entities. 
You can check withings binary sensors or tedee for different approaches how to achieve this. I would leave this for a follow up PR though.

Reply:
Thanks! That sounds like a good opportunity to add support for a reconfigure step too. Would it make sense to keep this as-is for now, or should I rework this into `_async_setup`?

Reply:
If you’re aiming for dynamic entities just move this to `_async_update` already